collect_app/src/main/java/org/odk/collect/android/activities/MainMenuActivity.java

```diff
@@ -116,6 +123,18 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.main_menu);
         initToolbar();
 
+        // Enable TLS v1.2 for Android 4.2.2
+        try {
+            ProviderInstaller.installIfNeeded(getApplicationContext());
```

Comment:
This updates a device's security provider and that raises some red flags for me. According to https://developer.android.com/training/articles/security-gms-provider, this can take time (max: 350ms) and since it's being called on the UI thread, the correct call is probably `installIfNeededAsync()`, not `installIfNeeded()`. 

Before merging, it'd be good to understand 
1. Could this call downloading a big update? If so, that's is a bad experience for our users in the field.
1. Does updating the provider have any downsides or is it very safe?